### PR TITLE
Build System

### PR DESCRIPTION
**Pull request by randomPoison**
- Turned the build settings into an asset file.
- Automatically name folders by version and target platform.
- Automatically generate build ID.

### CODE REVIEW

File: Assets/Editor/BuildUtils.cs [hunk 1]
```diff
@@ -1,33 +1,129 @@
...
+			case BuildTarget.StandaloneWindows64:
+				return ".exe";
+			case BuildTarget.StandaloneOSXIntel:
+			case BuildTarget.StandaloneOSXIntel64:
+			case BuildTarget.StandaloneOSXUniversal:
+				return ".app";
+			case BuildTarget.StandaloneLinux:
+			case BuildTarget.StandaloneLinux64:
+			case BuildTarget.StandaloneLinuxUniversal:
+				return ".LINUX"; // TODO what even is
```
Comment (brickman1444) [inline Assets/Editor/BuildUtils.cs:77]: lol

File: Assets/Editor/BuildUtils.cs [hunk 2]
```diff
@@ -1,33 +1,129 @@
...
+	 * @todo
+	 *     Create new asset in currently selected folder of the project view,
+	 *     rather than always placing them in the root Assets folder.
+	 *
+	 * @todo
+	 *     Check if asset already exists, because by default the AssetDatabase
+	 *     will overwrite an existing asset of the same name.
+	 */
+	[MenuItem( "Assets/Create/Build Settings" )]
+	public static void CreateBehaviorTreeAsset()
```
Comment (brickman1444) [inline Assets/Editor/BuildUtils.cs]: Typo on the name?

**Comment by brickman1444** [comment]
Done.

File: Assets/Editor/BuildUtils.cs (hunk 2 above)
Comment (randomPoison) [inline Assets/Editor/BuildUtils.cs]: ?_?

Woops, I got it.

File: Assets/Editor/BuildUtils.cs (hunk 1 above)
Comment (randomPoison) [inline Assets/Editor/BuildUtils.cs:77]: IKR? Just another one of the things that Unity inexplicably doesn't handle properly by default.

**Comment by cjacobwade** [comment]
Looks good

**Pull request merged**